### DISCUSSION TITLE
Video export is flickering, SVG export is fluent

### DISCUSSION THREAD

**Discussion by hugogonzalezcastro** [friction2d/friction#187] [Q&A]
Hello, I made several logo animations with characters scaling and moving. The animated SVG files that Friction export look great and fluent, but when I try to export a video, sometimes a character flickers in the video. It even happens when exporting an image sequence, and some individual images are with missing characters...  although the characters never hide in the SVG. Is there any way to fix this issue? It happenes to me with version 0.9.5 and also 0.9.6 RC1 and RC2.

Also, animating the "opacity" property, works perfect in the SVG, but sometimes doesn't fade properly when exporting video.

Apart from those, I found sometimes the "visibility range" freezes the objects (do not animate) in the SVG export, but not in the preview window.

Thank you for any help in advance!

**Comment by rodlie** [comment]
Hi, thanks for your feedback.

I have noticed flickering on some projects (usually related to opacity), using CPU-only solves the issue for me. 

I don't know where it happens in the code, so I have no proper solution.

> Also, animating the "opacity" property, works perfect in the SVG, but sometimes doesn't fade properly when exporting video.

> I found sometimes the "visibility range" freezes the objects (do not animate) in the SVG export, but not in the preview window.

Will check.

**Reply by rodlie** [reply]
> I found sometimes the "visibility range" freezes the objects (do not animate) in the SVG export, but not in the preview window.

Are you able to share a SVG (zip it and add attachment) where this happens?

**Comment by hugogonzalezcastro** [comment]
Hello, thank you very much for you answers!

I have prepared an ungly example with the issue elements only, removing a lot of parts of the original animation that were totally unrelated, to simplify: [FrictionBug.zip](https://github.com/user-attachments/files/15777327/FrictionBug.zip)

I include two versions of the same example, one using the visibility range for an object called "mask" that if moving as a mask between two characters. When the "visibility range" is active, the animation is working inside the program but not working when exporting the svg as a preview (that seems to be because the mask if not moving when exporting, and it is easy to check by changing the colour of the mask object). I also include the export in html I got. Also, I include the "expected" version, which doesn't use the "visibility range".

The flickering happened in the character "R" of the animated word: "WORK", and I do not animate the opacity for these characters...  I animate the opacity for the copyright symbol and the word "or". I also get some visibility issues with the "R" in the working window, which is weird, but I do not animate the opacity for those characters...

I think there might be a memory issue that might corrupt attributes sometimes, as that gets serialized on the file. I think a nice posibility could be **saving the Friction files in plain text format (xml, json,...) instead of a binary file** (or supporting both), so we could compare the projects with previous versions to check for any changing properties manually...  could be a nice way to detectd these kind of bugs.

I will check the options for "CPU-only solves" to see if that solves my issue too, thank you!

If I have more time in the future, I would like to download the source code myself, I have experience with C++ code and it would be really interesting. Also, it is a pity **Windows 8 and 7** are not officialy supported, as this program would work fine in old computers, and also works in Linux!

Thank you for any help!
